demo/agw/ScrolledThumbnail.py

```diff
@@ -0,0 +1,640 @@
+
+        self.SetIcon(images.Mondrian.GetIcon())
+        self.SetTitle("ScrolledThumbnail wxPython Demo ;-)")
+
+        self.statusbar = self.CreateStatusBar(2)
+        self.statusbar.SetStatusWidths([-2, -1])
+        # statusbar fields
+        statusbar_fields = [("ScrolledThumbnail Demo, Michael Eager @ 23 Oct 2020"),
+                            ("Welcome To wxPython!")]
+
+        for i in range(len(statusbar_fields)):
+            self.statusbar.SetStatusText(statusbar_fields[i], i)
+
+        self.SetMenuBar(self.CreateMenuBar())
+
+        splitter = wx.SplitterWindow(self, -1, style=wx.CLIP_CHILDREN |
+                                     wx.SP_3D | wx.WANTS_CHARS | wx.SP_LIVE_UPDATE)
+        self.panel = wx.Panel(splitter, -1)
+
+        sizer = wx.BoxSizer(wx.HORIZONTAL)
+
+        self.scroll = ScrolledThumbnail(splitter, -1, size=(400,300))
+
+        self.log = log
+
+        self.thumbsizer_staticbox = wx.StaticBox(self.panel, -1, "Thumb Style")
+        self.customsizer_staticbox = wx.StaticBox(self.panel, -1, "Thumb Customization")
+        self.optionsizer_staticbox = wx.StaticBox(self.panel, -1, "More Options")
+        self.dirsizer_staticbox = wx.StaticBox(self.panel, -1, "Directory Selection")
+        self.dirbutton = wx.Button(self.panel, -1, "Change Directory")
+        self.radiostyle1 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_NONE", style=wx.RB_GROUP)
+        self.radiostyle2 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_FULL")
+        self.radiostyle3 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_RECT")
+        self.radiostyle4 = wx.RadioButton(self.panel, -1, "THUMB_OUTLINE_IMAGE")
+        self.highlight = wx.CheckBox(self.panel, -1, "Highlight on pointing")
+        self.showfiles = wx.CheckBox(self.panel, -1, "Show file names")
+        self.enabledragging = wx.CheckBox(self.panel, -1, "Enable drag and drop")
+        self.setpopup = wx.CheckBox(self.panel, -1, "Set popup menu on thumbs")
+        self.setgpopup = wx.CheckBox(self.panel, -1, "Set global popup menu")
+        self.enabletooltip = wx.CheckBox(self.panel, -1, "Enable thumb tooltips")
+        self.textzoom = wx.TextCtrl(self.panel, -1, "1.4")
+        self.zoombutton = wx.Button(self.panel, -1, "Set zoom factor")
+        self.textthumbwidth = wx.TextCtrl(self.panel, -1, "96")
+        self.textthumbheight = wx.TextCtrl(self.panel, -1, "80")
+        self.thumbsizebutton = wx.Button(self.panel, -1, "Set thumbnail size (WxH)")
+        self.fontbutton = wx.Button(self.panel, -1, "Set caption font")
+        self.colourbutton = wx.Button(self.panel, -1, "Set selection colour")
+
+        self.radios = [self.radiostyle1, self.radiostyle2, self.radiostyle3,
+                       self.radiostyle4]
+        self.thumbstyles = ["THUMB_OUTLINE_NONE", "THUMB_OUTLINE_FULL", "THUMB_OUTLINE_RECT",
+                            "THUMB_OUTLINE_IMAGE"]
+
+        self.SetProperties()
+        self.DoLayout()
+
+        self.panel.SetSizer(sizer)
```

Comment:
In `DoLayout` you are adding the widgets to a sizer and doing: 
```
        self.panel.SetSizer(splitsizer)
```

And then in the code above you are replacing that sizer with another with nothing in it.

Reply:
This is in the original demo/agw/ThumbnailCtrl.py code.  

When I remove line 85, setting the unused sizer, the layout changes.  All of the spacing between the Change Directory button, the Thumb Style, and other options disappears.  I'm pretty confused about why this is happening.

Reply:
> This is in the original...

Yes, I know. But that doesn't mean that it was correct. 😉 

You can use the Widget Inspection Tool (F6 in the demo) to explore the widget hierarchy and sizers to help you figure out why things are not working as you expect.

Reply:
Also, since there is so much overlap between these two samples it would be good if you could refactor things such that the common code could be shared between them. For example, the panel full of widgets would be a good candidate for sharing. 

Or perhaps it could be reorganized so the whole (or most of) sample is common code, except for which thumbnail control is used. Then in each case the sample code can just pass the class to be used into the frame's `__init__` method.

Reply:
Of course it’s correct, I wrote it 😜

Reply:
I'll look into refactoring the demo code.  Given how ThumbnailCtrl is written, it may not be as simple or clean as you suggest.  But there is a lot of code in common.  

When I try the Widget Inspection Tool in the demo, all I see are the panels/sizers for main.py, not the ScrolledThumbnail demo code.

Reply:
> When I try the Widget Inspection Tool in the demo, all I see are the panels/sizers for main.py, not the ScrolledThumbnail demo code.

You may need to use the refresh tool in the toolbar to add the new frame to the WIT's list of windows/widgets.

Reply:
I removed the unused sizer.  The StaticBoxes around the controls were not rendered in the original ThumbnailCtrl.  Removing the unused sizer caused them to be rendered and this was the cause of the change in the layout.  I added spacers to fix the layout.  (I don't really understand why removing the sizer changed the layout, but lifes full of little mysteries.)

I moved all of the common code in the demos for ScrolledThumbnail and ThumbnailCtrl to a class ThumbDemoConfig. Both demos inherit from this class.  There are a few methods which are different between the two.

I added documentation to both demos and to the common widget code.